packages/next/src/server/lib/dedupe-fetch.ts

```diff
@@ -74,29 +83,35 @@ export function createDedupeFetch(originalFetch: typeof fetch) {
     }
 
     const cacheEntries = getCacheEntries(url)
-    let match
-    if (cacheEntries.length === 0) {
-      // We pass the original arguments here in case normalizing the Request
-      // doesn't include all the options in this environment.
-      match = originalFetch(resource, options)
-      cacheEntries.push(cacheKey, match)
-    } else {
-      // We use an array as the inner data structure since it's lighter and
-      // we typically only expect to see one or two entries here.
```

Comment:
you refactored away from the array impl but I think the original motivation still stands. It's more efficient than a dictionary object for lookups even if it is linear time b/c in practice there will be usually 0 or 1 entry to match not many and thus the overhead of the object creation and possible hidden class lookups is greater than just iterating a tick or two